### PR TITLE
changing pixel verification into visibility verification

### PR DESCRIPTION
**Pull request by justnpT**
The methods affected by the change:
MercuryTOCTest_001_TOCPresence_ListRedirection ()
MercuryTOCTest_003_RedirectionToHeaderDirectlyFromLink ()
MercuryTOCTest_004_RedirectionToHeaderFromCurrentPage ()
MercuryTOCTest_005_RedirectionToHeaderFromOtherPage ()

The tests were verfified on the functional level
The tests were verified on the local stability level

https://wikia-inc.atlassian.net/browse/XW-909

### CODE REVIEW

File: src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/mercury/TableOfContentPageObject.java [hunk 1]
```diff
@@ -54,16 +57,12 @@ public void clickOnTOCListElement(int index) {
...
-    JavascriptExecutor js = (JavascriptExecutor) driver;
-    String h2PosString =
-        js.executeScript(
-            "return Math.floor($('section.article-body h2').eq(" + index + ").offset().top)")
-            .toString();
-    int h2Pos = Integer.parseInt(h2PosString);
-    int windowYPos =
-        Integer.parseInt(js.executeScript("return $(window).scrollTop()").toString());
-    return h2Pos == windowYPos;
+  public boolean isUserMovedToSectionByIndex(int sectionIndex) {
```
Comment (tomasznapieralski) [inline src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/mercury/TableOfContentPageObject.java]: change `int` --> `string`

File: src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/mercury/TableOfContentPageObject.java [hunk 2]
```diff
@@ -54,16 +57,12 @@ public void clickOnTOCListElement(int index) {
...
+    String sectionIndexString = String.valueOf(sectionIndex);
```
Comment (tomasznapieralski) [inline src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/mercury/TableOfContentPageObject.java]: remove this

File: src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/mercury/TableOfContentPageObject.java [hunk 3]
```diff
@@ -54,16 +57,12 @@ public void clickOnTOCListElement(int index) {
...
+    By sectionBy = By.cssSelector(section_selector.replace("__index__", sectionIndexString));
```
Comment (tomasznapieralski) [inline src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/mercury/TableOfContentPageObject.java]: remove this

File: src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/mercury/TableOfContentPageObject.java [hunk 4]
```diff
@@ -54,16 +57,12 @@ public void clickOnTOCListElement(int index) {
...
+    WebElement section = driver.findElement(sectionBy);
```
Comment (tomasznapieralski) [inline src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/mercury/TableOfContentPageObject.java]: replate this line with:
`WebElement section = driver.findElement(By.cssSelector(section_selector.replace("__index__", sectionIndex)));`

Comment (tomasznapieralski) [inline src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/mercury/TableOfContentPageObject.java]: or with:
`WebElement section = driver.findElement(By.cssSelector("section.article-body h2[section=\" + sectionIndex + "\"]"));`

**Comment by tomasznapieralski** [comment]
CR done

File: src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/mercury/TableOfContentPageObject.java (hunk 4 above)
Comment (justnpT) [inline src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/mercury/TableOfContentPageObject.java]: I will use this alternative:
WebElement section = driver.findElement(By.cssSelector(section_selector.replace("__index__", sectionIndex)));

I can't use the second alternative, the css selector should be part of UI mapping (which is on top of the page component)

Comment (tomasznapieralski) [inline src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/mercury/TableOfContentPageObject.java]: this is simple `String` it is not UI mapping. Comparing two options:
1) `section_selector.replace("__index__", sectionIndex)`
2) `section.article-body h2[section=\"" + sectionIndex + "\"]"`

option 2) gives you chance to see how exactly string will looks like

Comment (justnpT) [inline src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/mercury/TableOfContentPageObject.java]: This is string that represents a css selector, so this should be part of the UI map:
http://www.seleniumhq.org/docs/06_test_design_considerations.jsp#ui-mapping

A UI Map is a mechanism that stores all the locators for a test suite in one place for easy modification when identifiers or paths to UI elements change in the AUT. The test script then uses the UI Map for locating the elements to be tested. Basically, a UI map is a repository of test script objects that correspond to UI elements of the application being tested.

https://code.google.com/p/selenium/wiki/PageFactory

**Pull request merged**